pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -95,6 +95,11 @@ def reset(self) -> None:
         assert self.batch_loop.optimizer_loop is not None
         self.is_last_batch = False
 
+        if self.restarting:
+            self.batch_progress.current.reset_on_restart()
+            self.scheduler_progress.current.reset_on_restart()
```

Comment:
can you put these with the `if not self.restarting below`? L106

Reply:
Done !

Reply:
Actually, I moved it up, It makes it weird to have 2 if one after the other. At least it is cleared like this.

Reply:
I didn't understand your comment

Reply:
Thoughts about this @awaelchli? need to be careful here because `_num_training_batches_reached` depends on the progress value

Reply:
No it is not an elif condition. It is a double if condition.